static/app/components/modals/dashboardWidgetQuerySelectorModal.spec.tsx

```diff
@@ -110,40 +106,33 @@ describe('Modals -> AddDashboardWidgetModal', function () {
       conditions: 'title:/organizations/:orgId/',
       id: '3',
     });
-    const wrapper = mountModal({initialData, widget: mockWidget});
-    await tick();
-    expect(wrapper.find('StyledInput').length).toEqual(3);
-    expect(wrapper.find('StyledInput').at(0).props().value).toEqual(
-      'title:/organizations/:orgId/performance/summary/'
-    );
-    expect(wrapper.find('StyledInput').at(1).props().value).toEqual(
-      'title:/organizations/:orgId/performance/'
-    );
-    expect(wrapper.find('StyledInput').at(2).props().value).toEqual(
-      'title:/organizations/:orgId/'
-    );
-    expect(wrapper.find('OpenInDiscoverButton').length).toEqual(3);
-    wrapper.unmount();
+    renderModal({initialData, widget: mockWidget});
+    const queryFields = screen.getAllByLabelText('Search events');
+    expect(queryFields).toHaveLength(3);
+
+    screen.getByDisplayValue('title:/organizations/:orgId/performance/summary/');
+    screen.getByDisplayValue('title:/organizations/:orgId/performance/');
+    screen.getByDisplayValue('title:/organizations/:orgId/');
   });
 
-  it('links user to the query in discover when a query is selected from the modal', async function () {
-    const wrapper = mountModal({initialData, widget: mockWidget});
-    await tick();
-    expect(wrapper.find('QueryContainer').find('Link').props().to).toEqual(
+  it('links user to the query in discover when a query is selected from the modal', function () {
+    renderModal({initialData, widget: mockWidget});
+    expect(
+      screen.getByLabelText('Open in Discover')?.parentElement?.getAttribute('href')
```

Comment:
ooof, this is gross

Does `getByRole` with button and `toHaveAttribute` not work?

Reply:
Yeah this is kind of gross haha, it's a button wrapped with a hyperlink so that's why I used the label to reach for the parent's href.

I just changed this to `getByRole` on the `link` and used `toHaveAttribute`, I forgot about that handy assertion. Thanks!